lib/schlepper/process.rb

```diff
@@ -75,32 +75,51 @@ def say(*); end
 
     private def process_one klass
       runner = klass.new
+      use_transaction = klass::USE_MIGRATION if defined? klass::USE_MIGRATION
 
       puts ''
       puts "Processing #{klass.name} from #{runner.owner}:"
       puts "#{runner.description}"
       puts ''
 
-      ActiveRecord::Base.transaction do
-        if runner.run
-          ActiveRecord::Base.connection.execute <<-SQL
-            INSERT INTO schlepper_tasks (version, owner, description, completed_at)
-            VALUES (#{runner.version_number}, #{ActiveRecord::Base.sanitize(runner.owner)}, #{ActiveRecord::Base.sanitize(runner.description)}, #{ActiveRecord::Base.connection.quote(Time.now.to_s(:db))});
-          SQL
-        else
-          puts "#{klass.name} ran without errors, but was not successful"
-          if runner.failure_message
-            puts "The resulting failure was: #{runner.failure_message}"
-          else
-            puts "The failure message was not set. Find #{runner.owner} to help investigate"
+      if use_transaction == false
+        status = run runner
+        log_error(klass.name, runner.failure_message, runner.owner) unless status
+      else
+        ActiveRecord::Base.transaction do
+          status = run runner
+          unless status
+            log_error(klass.name, runner.failure_message, runner.owner)
+            fail ActiveRecord::Rollback
           end
-          fail ActiveRecord::Rollback
         end
       end
 
       puts ''
       puts "Finished #{klass.name}"
       puts '~~~~~~~~~~~~~~~~~~~~~'
     end
+
+    private def run(runner)
```

Comment:
This is a bit confusing with this method `run` and the task method's `#run`. Perhaps `run_task_for`?